### PR TITLE
Use pytest env for pytest CI check

### PR DESCRIPTION
**Pull request by mpharrigan**

### CODE REVIEW

**Comment by fdmalone** [comment]
our pytest env is insufficient / out of date it seems

**Comment by mpharrigan** [comment]
sorry this is in progress to fix #517 among other issues.

This first commit gives the expected failure; I'll push a fix -- and we'll be left with the _other_ failures which then I'll think about how to fix :p

**Comment by mpharrigan** [comment]
ok. This is now ready. I changed the pytest job to run the runtime tests with runtime dependencies and added a new, separate job to test the dev dependencies. Previously: the dev environment was used for testing so badly configured factorized dependencies could sneak by causing #517

File: check/pytest-dev-tools [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Runs pytest on the repository.
```
Comment (fdmalone) [inline check/pytest-dev-tools:4]: runs pytest on dev_tools right?

File: check/pytest-dev-tools [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Usage:
+#     check/pytest [--flags for pytest] [file-paths-relative-to-repo-root]
```
Comment (fdmalone) [inline check/pytest-dev-tools:7]: fix docstring

**Review by fdmalone — CHANGES_REQUESTED** [review]
LGTM, just a nit about the docstring.

**Comment by mpharrigan** [comment]
ahh I had that commit locally but didn't push it because I didn't want to have to start the CI over :D

**Comment by fdmalone** [comment]
a [skip ci] might slide.

**Comment by mpharrigan** [comment]
is this good now @fdmalone ?

**Review by fdmalone — APPROVED** [review]

**Pull request merged**